packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -17,6 +20,56 @@ import deepmerge from 'deepmerge';
 import type { AbstractRpcService } from './abstract-rpc-service';
 import type { AddToCockatielEventData, FetchOptions } from './shared';
 
+/**
+ * Options for the RpcService constructor.
+ */
+export type RpcServiceOptions = {
+  /**
+   * A function that can be used to convert a binary string into a
+   * base64-encoded ASCII string. Used to encode authorization credentials.
+   */
+  btoa: typeof btoa;
+  /**
+   * The URL of the RPC endpoint to hit.
+   */
+  endpointUrl: URL | string;
+  /**
+   * An RPC service that represents a failover endpoint which will be invoked
+   * while the circuit for _this_ service is open.
+   */
+  failoverService?: AbstractRpcService;
+  /**
+   * A function that can be used to make an HTTP request. If your JavaScript
+   * environment supports `fetch` natively, you'll probably want to pass that;
+   * otherwise you can pass an equivalent (such as `fetch` via `node-fetch`).
+   */
+  fetch: typeof fetch;
+  /**
+   * A common set of options that will be used to make every request. Can be
+   * overridden on the request level (e.g. to add headers).
+   */
+  fetchOptions?: FetchOptions;
+  /**
+   * Options to pass to `createServicePolicy`. Note that `retryFilterPolicy` is
+   * not accepted, as it is overwritten. See {@link createServicePolicy}.
+   */
+  policyOptions?: Omit<CreateServicePolicyOptions, 'retryFilterPolicy'>;
+};
+
+/**
+ * The maximum number of times that a failing service should be re-run before
+ * giving up.
+ */
+export const DEFAULT_MAX_RETRIES = 4;
```

Comment:
We were already using these defaults, I just pulled them out for consistency with `createServicePolicy`.